### PR TITLE
Show "starting" rather than "idle" status when changing kernel via `SessionContext.changeKernel()`

### PR DESCRIPTION
**Pull request by Zsailer**
`SessionContext.kernelDisplayStatus` returns `'restarting'` when a restart is in flight and `'terminating'` when a shutdown is in flight, but nothing equivalent for the change-kernel window. During `_changeKernel()` on an existing session, `session.changeKernel(model)` is awaited (a PATCH round-trip plus new kernel provisioning), and `session.kernel` is still the previous kernel throughout that window. The `!kernel && _pendingKernelName` branch of the display getter is skipped because `kernel` is truthy, so the getter falls through to the previous kernel's `status`, typically `'idle'`.

The result is that when a user picks a different kernel from the kernel picker, the indicator keeps reporting the old kernel's state (usually `'idle'`) through the entire provisioning window, then jumps straight to the new kernel's post-`kernel_info_reply` `'idle'`. Any transient `'starting'` emitted by the new kernel is invisible.

Track an `_isChangingKernel` flag alongside `_isRestarting`, set it around the `session.changeKernel(model)` call, and return `'starting'` from the display getter while it's set. Also emit `statusChanged` at the boundaries so consumers repaint.

### CODE REVIEW

**Comment by Zsailer** [comment]
For context, this bug is really apparent when [pending kernels](https://jupyter-client.readthedocs.io/en/stable/pending-kernels.html) are enabled. When a kernel takes _any_ time to provision and get started, the UX in JupyterLab will stay e.g. "idle", while the kernel is really in a _starting_ state.

**Comment by Zsailer** [comment]
Reproducing takes a bit of work. You can mock a slow starting kernel using a custom Kernel Manager or Provisioner and add an arbitrarily long sleep in the start kernel logic. Then, enable pending kernels.

**Comment by krassowski** [comment]
@Zsailer is the updated title acurrate? Would you like to backport this to `4.6.x`?

File: packages/apputils/src/sessioncontext.tsx [hunk 1]
```diff
@@ -950,12 +954,17 @@ export class SessionContext implements ISessionContext {
...
       try {
         await this._session.changeKernel(model);
         return this._session.kernel;
       } catch (err) {
         void this._handleSessionError(err);
         throw err;
+      } finally {
+        this._isChangingKernel = false;
+        this._statusChanged.emit(this._session?.kernel?.status || 'unknown');
       }
```
[the remark this answers is not in the document]
Comment (krassowski) [reply packages/apputils/src/sessioncontext.tsx:965]: Reading the code, it appears that it will emit `'unknown'` twice indeed.

We probably should add a unit test that mocks rejection on `changeKernel` and asserts that the `unknown` status is emitted once.

**Review by krassowski — APPROVED** [review]
Tested on Binder looks good. Copilot flagged and edge case where on rejection the status changed signal gets emitted twice. It is not blocking the merge, but would probably be good to fix and cover with a test case. Let me know if you would like to address it here or just merge as-is

**Comment by Zsailer** [comment]
Thanks so much @krassowski!
